hera_cal/abscal_funcs.py

```diff
@@ -1059,201 +1213,124 @@ def __eq__(self, B2):
         if np.isclose(self.len, B2.len, atol=tol):
             # check x, y, z
             equiv = bool(reduce(operator.mul, map(lambda x: np.isclose(*x, atol=tol), zip(self.bl, B2.bl))))
+            dot = np.dot(self.unit, B2.unit)
             if equiv:
                 return True
             # check conjugation
-            elif np.isclose(np.arccos(np.dot(self.unit, B2.unit)), np.pi, atol=tol/self.len):
+            elif np.isclose(np.arccos(dot), np.pi, atol=tol/self.len) or (dot < -1.0):
                 return 'conjugated'
             # else return False
             else:
                 return False
         else:
             return False
 
-def smooth_data(Xtrain, ytrain, Xpred, kind='gp', n_restart=3, degree=1, ls=10.0, return_model=False):
+def match_red_baselines(model, model_antpos, data, data_antpos, tol=1.0, verbose=True):
```

Comment:
This function (and the baseline class) seem too general and complicated. More importantly, we need to figure out how to properly propagate weights.

Reply:
agreed